linetools/isgm/utils.py

```diff
@@ -463,53 +471,65 @@ def group_coincident_compoments(comp_list, output_type='list'):
     if output_type not in ['list', 'dict', 'dictionary']:
         raise ValueError("`output_type` must be either 'list' or 'dict'.")
 
-    # the first extreme case is that all components are independent
-    # of each other, in which case we have the following output shape
-    out = [[] for kk in range(len(comp_list))]
-
-    for ii in range(len(comp_list)):
-        comp_ii = comp_list[ii]
-        # only append if ii does not belong to a previous round
-        switch = 0
-        for kk in range(len(out[:ii])):
-            if ii in out[kk]:
-                switch = 1
-                break
-        if switch == 1:
-            pass
+    ### We first want to identify and group all blended lines
+    ### Sort them by observed wavelength to do this
+    lst=[]
+    compnos=[]
+    for ii,comp in enumerate(comp_list):
+        lst.extend(comp._abslines)
+        compnos.extend([ii]*len(comp._abslines))
+    lst=np.array(lst)
```

Comment:
Why work in absline space rather than component space? There is already implemented a function called `coincident_compoments` that checks whether two components have 1 or more overlapping abslines within them.